fiona/ogrext.pyx

```diff
@@ -1342,13 +1349,6 @@ cdef class WritingSession(Session):
                         record['geometry']['type'],
                         collection.schema['geometry'] ))
 
-            # Validate against collection's schema to give useful message
-            if set(record['properties'].keys()) != schema_props_keys:
-                raise SchemaError(
-                    "Record does not match collection schema: %r != %r" % (
-                        record['properties'].keys(),
-                        list(schema_props_keys) ))
```

Comment:
this duplicates lines 1338 -- 1343, with a different exception type, so i would expect this code to be unreachable.  perhaps these two blocks are an unintended result of a merge